src/experimental/safe_html_transform/transforms/exp_safe_html.py

```diff
@@ -293,24 +293,22 @@ def convert(self, orig, data, **kwargs):
         else:
             # append html tag to create a dummy parent for the tree
             html = "<html>%s</html>" % orig
+            parser = etree.HTMLParser()
+            tree = etree.parse(StringIO(html), parser)
+            for element in tree.getiterator():
+                if(element.tag == 'h3' or element.tag == 'h4' or element.tag == 'h5' or element.tag == 'h6' or element.tag == 'div'):
+                    element.tag = 'p'
+                if(element.tag == "html" or element.tag == "body" or element.tag == "script"):
+                    etree.strip_tags(tree, element.tag)
+            result = etree.tostring(tree.getroot(), pretty_print=True, method="html")
             NASTY_TAGS = frozenset(['style', 'script', 'object', 'applet', 'meta', 'embed'])  # noqa
             cleaner = HTMLParser(kill_tags=NASTY_TAGS, page_structure=False, safe_attrs_only=False)
-            safe_html = fragment_fromstring(cleaner.clean_html(html))
+            safe_html = fragment_fromstring(cleaner.clean_html(result))
```

Comment:
You can pass your ``tree`` object here, you don't need to convert it to a string first.